File: backend/group/group.go
package group

import "time"

// Strategy 是分组的请求路由策略，每个分组必配其一。
type Strategy string

const (
	StrategyManual     Strategy = "manual"      // 手动指定：只打激活项
	StrategyRoundRobin Strategy = "round_robin" // 轮询：组内轮流
	StrategyFailover   Strategy = "failover"    // 故障转移：按 priority 顺序，失败换下一个
)

// Valid 校验策略枚举。
func (s Strategy) Valid() bool {
	switch s {
	case StrategyManual, StrategyRoundRobin, StrategyFailover:
		return true
	default:
		return false
	}
}

// Group 是客户端模型名称（对外暴露的名字），内部聚合多个渠道模型。
type Group struct {
	ID           int64       `gorm:"primaryKey" json:"id"`
	Name         string      `gorm:"unique;not null" json:"name"` // 对外模型名
	Strategy     Strategy    `gorm:"not null;default:manual" json:"strategy"`
	ActiveItemID int64       `json:"active_item_id"` // manual 策略下当前激活项 ID
	Items        []GroupItem `gorm:"foreignKey:GroupID" json:"items,omitempty"`
	CreatedAt    time.Time   `json:"created_at"`
	UpdatedAt    time.Time   `json:"updated_at"`
}

// GroupItem 是分组内的一个模型项，引用 model 包中的模型（用 ModelID 关联）。
type GroupItem struct {
	ID       int64 `gorm:"primaryKey" json:"id"`
	GroupID  int64 `gorm:"not null;uniqueIndex:idx_group_model" json:"group_id"`
	ModelID  int64 `gorm:"not null;uniqueIndex:idx_group_model" json:"model_id"` // 指向 model.Model
	Priority int   `json:"priority"`                                             // failover 按升序尝试；round_robin 无意义
}

File: web/src/renderer/api/group.js
import request from './request';

/**
 * 分组：对外模型名，聚合多个渠道模型，必配路由策略
 * strategy: manual | round_robin | failover
 * 分组项：{ id, group_id, model_id, priority }
 */

/** 分组列表（含分组项，按 priority 排序） */
export function listGroups() {
  return request.get('/groups');
}

/** 分组详情 */
export function getGroup(id) {
  return request.get(`/groups/${id}`);
}

/** 创建分组；可同时传初始分组项 { name, strategy, items?: [{model_id, priority?}] } */
export function createGroup(data) {
  return request.post('/groups', data);
}

/** 更新分组基础字段 { name?, strategy?, active_item_id? } */
export function updateGroup(id, data) {
  return request.put(`/groups/${id}`, data);
}

/** 删除分组（级联删除全部分组项） */
export function deleteGroup(id) {
  return request.delete(`/groups/${id}`);
}

/** 向分组添加模型 { model_id, priority? } */
export function addGroupItem(groupId, data) {
  return request.post(`/groups/${groupId}/items`, data);
}

/** 更新分组项（priority） */
export function updateGroupItem(groupId, itemId, data) {
  return request.put(`/groups/${groupId}/items/${itemId}`, data);
}

/** 从分组移除模型 */
export function deleteGroupItem(groupId, itemId) {
  return request.delete(`/groups/${groupId}/items/${itemId}`);
}
